### PR TITLE
This is supposed to make self signed certs work with the calendar module

### PR DESCRIPTION
**Pull request by codac**
Many people use Own-/Nextcloud together witht he https protocol. This is supposed to make self-signed certificates work with the calendar module and fix the issue #466.

### CODE REVIEW

**Comment by rejas** [comment]
Please add a CHANGELOG to your PR and make sure that all tests run fine, as well as all files ar elinted correctly:
"npm run test"
"npm run lint"

File: modules/default/calendar/calendarfetcher.js [hunk 1]
```diff
@@ -25,9 +25,10 @@ const moment = require("moment");
  * @param {number} maximumNumberOfDays The maximum number of days an event should be in the future.
  * @param {object} auth The object containing options for authentication against the calendar.
  * @param {boolean} includePastEvents If true events from the past maximumNumberOfDays will be fetched too
+ * @param {boolean} selfSignedCert If true, the server certificate is not verified against the list of supplied CAs.
  * @class
  */
-const CalendarFetcher = function (url, reloadInterval, excludedEvents, maximumEntries, maximumNumberOfDays, auth, includePastEvents) {
+const CalendarFetcher = function (url, reloadInterval, excludedEvents, maximumEntries, maximumNumberOfDays, auth, includePastEvents, selfSignedCert) {
```
Comment (rejas) [inline modules/default/calendar/calendarfetcher.js:31]: so, where is that parameter now passed?

Comment (codac) [reply modules/default/calendar/calendarfetcher.js:31]: in the node_helper.js

**Comment by codac** [comment]
Sorry guys, I'm all new to github, It took me a couple of days to finish the changes but now I think it should work.
Changes.md has been changed, Tests seem to be fine as well.
If there is anything lef to do please support me and tell me what it is.

**Comment by rejas** [comment]
> Sorry guys, I'm all new to github, It took me a couple of days to finish the changes but now I think it should work.
> Changes.md has been changed.
> If there is anything lef to do please support me and tell me what it is.

No worries, everybody has started sometimes :-) First run "npm run lint:prettier" to automatically fix style issues and commit 
them, so that the tests turn green hopefully.

**Review by rejas — APPROVED** [review]

**Comment by codac** [comment]
> 
> 
> > Sorry guys, I'm all new to github, It took me a couple of days to finish the changes but now I think it should work.
> > Changes.md has been changed.
> > If there is anything lef to do please support me and tell me what it is.
> 
> No worries, everybody has started sometimes :-) First run "npm run lint:prettier" to automatically fix style issues and commit
> them, so that the tests turn green hopefully.

Thank you for your understanding!
How can I fix the last error "codecov/project - 39.31% (-0.02%) compared to 09ddd3d"?

**Comment by rejas** [comment]
>  How can I fix the last error "codecov/project - 39.31% (-0.02%) compared to [09ddd3d](https://github.com/MichMich/MagicMirror/commit/09ddd3d92545eafeebabe0c0e9ac4ffb231a748c)"?

For that you would have to write a test for your feature. That is a little more advanced. I myself wouldn't know how to write one for your case without looking up a solution :-) 

But your PR could be merged even if that codecov stuff fails if nobody else has any comments?

**Comment by MichMich** [comment]
Thanks for your contribution @codac! I really appreciate it!

**Comment by khassel** [comment]
@codac meanwhile we replaced the `request` package with `node-fetch`, which also could affect your changes for self signed certs.

So it would be helpful if you can test if this is still working when using the newest `develop` branch of this repo, I have not test case for this. Thanks!

**Comment by codac** [comment]
> 
> 
> @codac meanwhile we replaced the `request` package with `node-fetch`, which also could affect your changes for self signed certs.
> 
> So it would be helpful if you can test if this is still working when using the newest `develop` branch of this repo, I have not test case for this. Thanks!

@khassel My tests were successful, seems to work.

**Comment by b-reich** [comment]
No docs about this here? 
https://docs.magicmirror.builders/modules/calendar.html#using-the-module

**Comment by rejas** [comment]
You are right @b-reich Do you want to contribute with a PR adding the missing config option?

**Pull request merged**